extensions/github/src/components/ProjectActions.tsx

```diff
@@ -0,0 +1,112 @@
+import { Action, ActionPanel, Icon, Toast, confirmAlert, showToast } from "@raycast/api";
+import { MutatePromise } from "@raycast/utils";
+
+import { getGitHubClient } from "../api/githubClient";
+import { ProjectFieldsFragment } from "../generated/graphql";
+import { getErrorMessage } from "../helpers/errors";
+import { hasRequiredScopes } from "../helpers/scopes";
+import { useMyProjects } from "../hooks/useMyProjects";
+
+type ProjectActionsProps = {
+  project: ProjectFieldsFragment;
+  mutateList?: MutatePromise<ProjectFieldsFragment[] | undefined> | ReturnType<typeof useMyProjects>["mutate"];
+  mutateDetail?: MutatePromise<ProjectFieldsFragment>;
+  children?: React.ReactNode;
+};
+
+export default function ProjectActions({ project, children, mutateList, mutateDetail }: ProjectActionsProps) {
+  const { github, octokit } = getGitHubClient();
+
+  const canClientActions = project.viewerCanClose || project.viewerCanReopen || project.viewerCanUpdate;
+
+  async function mutate() {
+    if (mutateList) {
+      await mutateList();
+    }
+
+    if (mutateDetail) {
+      await mutateDetail();
+    }
+  }
+
+  async function setProjectClosed(state: boolean) {
+    try {
+      await showToast({ style: Toast.Style.Animated, title: state ? `Closing project` : `Reopening project` });
+      const hasProjectScopes = await hasRequiredScopes(["project", "project:write"], octokit);
+
+      if (!hasProjectScopes) {
+        await showToast({
+          style: Toast.Style.Failure,
+          title: `Missing Scopes`,
+          message: `You need the "project" or "project:write" scopes to close or reopen a project. Please re-authenticate.`,
+        });
+        return;
+      }
+
+      await github.changeProjectStatus({
+        projectId: project.id,
+        closed: state,
+      });
+
+      await mutate();
+
+      await showToast({
+        style: Toast.Style.Success,
+        title: `Project "${project.title}" is now ${state ? "closed" : "reopened"}`,
+      });
+    } catch (error) {
+      await showToast({
+        style: Toast.Style.Failure,
+        title: `Failed ${state ? "closing" : "reopening"} project`,
+        message: getErrorMessage(error),
+      });
+    }
+  }
+
+  return (
+    <ActionPanel title={`Project: ${project.title}`}>
+      {children}
+
+      <Action.OpenInBrowser title="Open in GitHub" icon={Icon.Globe} url={project.url} />
+
+      {canClientActions ? (
+        <ActionPanel.Section>
+          {project.viewerCanClose && project.viewerCanUpdate && !project.closed ? (
+            <Action
```

Comment:
You can make this action [destructive](https://developers.raycast.com/api-reference/user-interface/actions#action.style) to make it a bit more scannable.